### PR TITLE
Implement schema v2 analyze workflow

### PR DESCRIPTION
**Pull request by anmorgunov**
why: PR5 needs the v2 score output to produce benchmark metrics instead of leaving users to hand-roll Solv-N, MRR, and reconstruction counts. The Solv-N bracket also needs a stable metric label so aggregate metrics are not ambiguous across task constraints.

what changed: added v2 AnalysisReport/MetricSummary models, pure analysis metrics with bootstrap CIs, the analyze workflow, task-derived metric labels, route-depth strata, and example-script logging. The bootstrap distribution helper is now typed generically so v2 can reuse it without old-schema adapters.

risk: metric semantics are the important boundary. Solv/MRR metrics use Evaluation.metric_label, Top-K reconstruction is emitted only when acceptable routes exist, and reconstruction ranking filters to task-satisfying candidates first. Default route-depth stratification uses primary acceptable route depth, falling back to route_depth constraints when acceptable routes are absent.

verification: uv run ruff check src/retrocast/v2 tests/v2 scripts/lib-example-use.py src/retrocast/metrics/bootstrap.py; uv run pytest tests/v2 tests/metrics/test_bootstrap.py; git diff --check.

follow-ups: CLI/IO wiring remains PR6 per the schema-v2 plan.

---
<a href="https://app.blacksmith.sh/ischemist/codesmith/project-procrustes/pr/114"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1782609423&installation_id=125602297&pr_number=114&repository=ischemist%2Fproject-procrustes&return_to=https%3A%2F%2Fgithub.com%2Fischemist%2Fproject-procrustes%2Fpull%2F114&signature=ea92f93af80f2a8a59998978875c59124c02c4085d078e796e298b3fab3712a1"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith</code> with what you need. Autofix is disabled.</sup>

## Release Notes

* **New Features**
  * Added `metric_label` field to tasks for customized metrics reporting
  * New `analyze()` function computes evaluation metrics with 95% confidence intervals
  * Metrics can be stratified by route depth or custom criteria

* **Documentation**
  * Updated schema design documentation with metric label field guidance

* **Tests**
  * Added comprehensive test coverage for metrics analysis and stratification functionality

[Review Change Stack](https://app.coderabbit.ai/change-stack/ischemist/project-procrustes/pull/114?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<h3>Greptile Summary</h3>

This PR implements the v2 analyze workflow, adding `AnalysisReport`/`MetricSummary` models, bootstrap-CI metric computation (Solv-N rate, MRR, top-k reconstruction), task-derived metric labels, and route-depth stratification on top of the existing v2 scoring pipeline.

- **New `analyze()` function** (`workflow/analyze.py`) aggregates `Evaluation` targets into global and per-stratum metrics; stratification defaults to primary acceptable route depth, falling back to the `route_depth` constraint.
- **`Task.derived_metric_label()`** derives a stable label from effective constraints (stock name, leaf, depth) used to namespace metric keys (e.g. `solv_0[buyables+depth]_rate`).
- **`bootstrap.py` generalised** from `list[TargetEvaluation]` to `Sequence[T]`, allowing v2 to reuse it without adapters; `rank ≥ 1` is now enforced on both `Candidate` and `ScoredCandidate`.

<h3>Confidence Score: 4/5</h3>

Safe to merge with one metric-correctness concern in the default stratification path.

The `_default_route_depth_stratum` fallback uses `f"depth {route_depth}"` directly, where `route_depth` can be a string literal. This produces stratum keys like `"depth short"` that co-exist with numeric `"depth 2"` keys in the same `AnalysisReport.by_stratum` dict, silently inflating stratum counts and making cross-stratum comparison misleading. All other logic — bootstrap CI computation, metric labelling, rank-zero guard, reconstruction filtering — looks correct.

src/retrocast/v2/workflow/analyze.py — specifically the `_default_route_depth_stratum` fallback branch

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| src/retrocast/v2/workflow/analyze.py | New `analyze()` workflow; `_default_route_depth_stratum` can silently mix string constraint labels ("depth short") with numeric measured-depth strata ("depth 2") when `route_depth` is a string literal |
| src/retrocast/v2/metrics/analysis.py | New metric computation helpers; rank ≥ 1 is now enforced upstream so the MRR 1/rank calculation is safe; bootstrap CIs via `_summarize_values` look correct |
| src/retrocast/v2/models/task.py | Adds `metric_label` field and `derived_metric_label()` / `effective_constraints()` helpers; label derivation logic is consistent with docs and tests |
| src/retrocast/v2/models/evaluation.py | Adds `metric_label` to `Evaluation` (defaulting to "task"); enforces `rank ≥ 1` on `ScoredCandidate` |
| src/retrocast/metrics/bootstrap.py | Generalises `get_bootstrap_distribution` from `list[TargetEvaluation]` to `Sequence[T]`; backward-compatible with existing v1 callers |
| src/retrocast/v2/workflow/score.py | Minor refactor: replaces inline `constraints.get(...)` with `effective_constraints()` and propagates `derived_metric_label()` into the returned `Evaluation` |
| src/retrocast/v2/models/analysis.py | New `MetricSummary` and `AnalysisReport` Pydantic models; straightforward and well-scoped |
| tests/v2/workflow/test_analyze.py | Comprehensive test coverage for all new metrics and stratification paths; edge cases (no acceptable routes, task-satisfaction filtering for top-k) are well covered |

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    E[Evaluation] --> A[analyze]
    A --> ST[summarize_targets\nglobal metrics]
    A --> STRAT{stratify_by?}
    STRAT -- custom fn --> CS[custom stratum key]
    STRAT -- None --> DR[_default_route_depth_stratum]
    DR -- acceptable_routes present --> RD[depth = acceptable_routes 0 .depth]
    DR -- no acceptable_routes --> CD[depth = effective_constraints.route_depth]
    CD -- int --> SK2[stratum 'depth N']
    CD -- string literal --> SK3[stratum 'depth short/medium/long']
    RD --> SK1[stratum 'depth N']
    CS --> SS[summarize_targets\nper-stratum metrics]
    SK1 --> SS
    SK2 --> SS
    SK3 --> SS
    ST --> AR[AnalysisReport]
    SS --> AR
    ST --> SV[_summarize_values\nbootstrap CI]
    SS --> SV
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 1 code review issue. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 1
src/retrocast/v2/workflow/analyze.py:49-55
**Mixed stratum key types when `route_depth` is a string constraint**

`TaskConstraints.route_depth` is typed `int | Literal["short", "medium", "long"] | None`. When a target has no `acceptable_routes` and its `route_depth` is a string (e.g. `"short"`), this function produces a stratum key of `"depth short"`. Meanwhile, targets whose depth is measured from `acceptable_routes[0].depth()` always produce numeric keys like `"depth 2"`.

These two cases end up as sibling keys in `AnalysisReport.by_stratum`. A target whose constraint says `route_depth="short"` might in practice have the same depth as a target that measured `depth=2`, but they land in different strata with no indication to the caller that the keys carry incomparable semantics (constraint label vs measured depth). This silently inflates the stratum count and can lead to incorrect per-depth aggregations.

`````

</details>

<sub>Reviews (2): Last reviewed commit: ["Address analyze review comments"](https://github.com/ischemist/project-procrustes/commit/4f2c1145c5870a15fe911bac0b98ef3349d24132) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=34496244)</sub>

### CODE REVIEW

File: src/retrocast/v2/metrics/analysis.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+def _mrr_solv(targets: Sequence[TargetResult], tier: Tier, *, n_boot: int, seed: int) -> MetricSummary:
+    values = []
+    for target in targets:
+        candidates = sorted(target.candidates, key=lambda candidate: candidate.rank)
+        reciprocal_rank = 0.0
+        for candidate in candidates:
+            if candidate.satisfies_solv(tier):
+                reciprocal_rank = 1.0 / candidate.rank
+                break
```
[the remark this answers is not in the document]
Comment (anmorgunov) [reply src/retrocast/v2/metrics/analysis.py:54]: @gemini-code-assist go back to python jail, we construct the Candidate, trust our code
